enumerables.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    array = []
+    my_each { |i| array.push(i) if yield(i) }
+    array
+  end
+
+  def my_all
+    my_each { |i| return false unless yield i }
+    true
+  end
+
+  def my_any
+    my_each { |i| return true if yield i }
+    false
+  end
+
+  def my_none
+    my_each { |i| return true unless yield i }
+    false
+  end
+
+  def my_count
+    count = 0
+    my_each { |i| count += 1 if yield i }
+    count
+  end
+
+  #   def my_map
+  #     array = []
+  #     my_each { |i| array.push yield i }
+  #   end
+  def my_map(&procs)
+    array = []
+    my_each { |n| array << (procs.nil? ? yield(n) : procs.call(n)) }
+    array
+  end
+
+  def my_inject(method)
```

Comment:
#### `REQUIRED`

### :bug: Let's add a case for when a block is given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-inject)

```ruby
irb(main):004:0> [1,2,3,4,5].inject{ |sum, n| sum + n }
=> 15
irb(main):005:0> [1,2,3,4,5].my_inject{ |sum, n| sum + n }
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):5
        1: from /../enumerables.rb:57:in `my_inject'
ArgumentError (wrong number of arguments (given 0, expected 1))
```